test/type/explicit_existential.swift

```diff
@@ -331,3 +331,15 @@ func testEnumAssociatedValue() {
     case c3((P) -> Void)
   }
 }
+
+// https://github.com/apple/swift/issues/58920
+typealias Iterator = any IteratorProtocol
+var example: any Iterator = 5 // expected-error{{redundant 'any' has no effect on existential type 'Iterator' (aka 'IteratorProtocol')}} {{14-18=}} 
```

Comment:
Seems this is actually `redundant 'any' has no effect on existential type 'Iterator' (aka 'any IteratorProtocol')` on 5.7, ie. main has `aka 'IteratorProtocol'` and 5.7 has `aka 'any IteratorProtocol'`. @hborla which is expected here?

Reply:
The `any` is expected. Printing `any` in the "aka" for typealiases was fixed by @AnthonyLatsis after the original patch for this diagnostic, and it has already been cherry-picked to 5.7: https://github.com/apple/swift/pull/59155

The original fix on main updated this test: https://github.com/apple/swift/pull/59147/files#diff-6784c46f13d1b6152029a9c875e5a456b7ad5cdc65328307b327c485d892cdbaR337

Reply:
That's... odd. I thought I checked on main. Must have been in the wrong folder 😅. Thanks, I'll update this PR.